core/src/main/java/fr/sncf/osrd/new_infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -0,0 +1,173 @@
+package fr.sncf.osrd.new_infra.implementation.tracks.undirected;
+
+import static fr.sncf.osrd.new_infra.api.tracks.undirected.TrackEdge.TRACK_OBJECTS;
+import static fr.sncf.osrd.new_infra.api.tracks.undirected.TrackObject.TrackObjectType;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.graph.ImmutableNetwork;
+import com.google.common.graph.NetworkBuilder;
+import fr.sncf.osrd.infra.InvalidInfraException;
+import fr.sncf.osrd.new_infra.api.tracks.undirected.*;
+import fr.sncf.osrd.railjson.schema.infra.RJSInfra;
+import fr.sncf.osrd.railjson.schema.infra.RJSSwitch;
+import fr.sncf.osrd.railjson.schema.infra.RJSSwitchType;
+import fr.sncf.osrd.railjson.schema.infra.RJSTrackSection;
+import fr.sncf.osrd.railjson.schema.infra.trackobjects.RJSRouteWaypoint;
+import fr.sncf.osrd.utils.graph.EdgeEndpoint;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class UndirectedInfraBuilder {
+
+    /** Constructor */
+    private UndirectedInfraBuilder() {
+        builder = NetworkBuilder
+                .directed()
+                .immutable();
+    }
+
+    /** Creates a TrackInfra from a railjson infra */
+    public static TrackInfra parseInfra(RJSInfra infra) {
+        return new UndirectedInfraBuilder().parse(infra);
+    }
+
+    private final HashMap<String, TrackNode> beginEndpoints = new HashMap<>();
+    private final HashMap<String, TrackNode> endEndpoints = new HashMap<>();
+    private final ImmutableNetwork.Builder<TrackNode, TrackEdge> builder;
+
+    /** Parse the railjson to build an infra */
+    private TrackInfra parse(RJSInfra infra) {
+        // Creates switches
+        var switchTypeMap = new HashMap<String, RJSSwitchType>();
+        for (var rjsSwitchType : infra.switchTypes)
+            switchTypeMap.put(rjsSwitchType.id, rjsSwitchType);
+        var switches = new ImmutableMap.Builder<String, Switch>();
+        for (var s : infra.switches) {
+            switches.put(s.id, parseSwitch(s, switchTypeMap));
+        }
+
+        // Create remaining links
+        for (var link : infra.trackSectionLinks) {
+            var newNode = new InfraTrackNode.Joint();
+            addNode(link.src.track.id.id, link.src.endpoint, newNode);
+            addNode(link.dst.track.id.id, link.dst.endpoint, newNode);
+        }
+
+        var trackSectionsByID = new HashMap<String, TrackSection>();
+        for (var track : infra.trackSections) {
+            var newTrack = makeTrackSection(track);
+            trackSectionsByID.put(newTrack.getID(), newTrack);
+        }
```

Comment:
I thought about it again and I think `navigability` field does not make sense. We can define a one way track section using routes instead of link navigability.

Reply:
I was puzzled by how to integrate this. I agree that using routes make more sense, we could remove the field from railjson